### PR TITLE
[GP-17] Create top level elements

### PR DESCRIPTION
**Pull request by dovaaa**

### CODE REVIEW

**Comment by dovaaa** [comment]
I created some Custom elements, this is subject to change under changes to slate-types.d.ts, Do not merge yet

File: src/pages/editor/components/slate/slate-types.d.ts [hunk 1]
```diff
@@ -77,6 +77,44 @@ export type PageElement = {
   padding: PaddingProps;
 }
 
+export type QuestionElement =
```
Comment (KareemMAX) [inline src/pages/editor/components/slate/slate-types.d.ts:80]: I'm thinking `QuestionElement` should be structured differently, as it includes a question part/heading and an answer part.
So, `AnswerAreaElement` is not a valid alias, in fact `QuestionElement` as a whole should be renamed to something that represent a heading as I think that is what you wanna do. What do you think?

File: src/pages/editor/components/slate/slate-types.d.ts [hunk 2]
```diff
@@ -77,6 +77,44 @@ export type PageElement = {
...
+  | MultipleChoiceQuestionElement
+  | MathQuestionElement
+  | CompareQuestionElement
+  | TrueFalseQuestionElement
+  | EssayQuestionElement
+  | AnswerAreaElement
+
+export type MultipleChoiceQuestionElement = {
```
Comment (KareemMAX) [inline src/pages/editor/components/slate/slate-types.d.ts:88]: I think here we should denote an array of possible answers.

**Review by KareemMAX — CHANGES_REQUESTED** [review]

**Comment by dovaaa** [comment]
[[GP-17] Create top level elements #10](https://trello.com/c/0gaOIZ6f/2-gp-17-create-top-level-elements-10)

**Comment by JonathanGhaly** [comment]
Can I Contribute?

**Comment by CatB1t** [comment]
Sure @JonathanGhaly

**Pull request closed without merging**